app/components/check-box.js

```diff
@@ -0,0 +1,28 @@
+import Ember from "ember";
+
+export default Ember.Component.extend({
+  attributeBindings: ["type", "value"],
+  tagName: "input",
+  type: "checkbox",
+  checked: false,
+  selection: [],
+
+  didInsertElement() {
+    var isCurrentRole =
```

Comment:
Components should be generic and reusable, so better not use the term `role` at all. Maybe substitute `isCurrentRole` by `isChecked` ?